app/reports/page.tsx

```diff
@@ -0,0 +1,132 @@
+import ReportCard from "@/components/reports/report-card";
+import { siteConfig } from "@/config/site";
+import { getNumberOfContributors } from "@/lib/directus";
+import { fetchReports } from "@/lib/impact-reports";
+import type { Report } from "@/types";
+import Link from "next/link";
+// import { useMemo } from "react";
+
+// export const meta: MetaFunction = () => {
+// 	return [
+// 		{ title: "VoiceDeck" },
+// 		{ name: "description", content: "Welcome to VoiceDeck!" },
+// 	];
+// };
+
+// interface IPageData {
+// 	reports: Report[];
+// 	numOfContributors: number;
+// }
+
+// export const loader: LoaderFunction = async ({ request }) => {
+// 	try {
+// 		const reports = await fetchReports();
+// 		const numOfContributors = await getNumberOfContributors();
+
+// 		return {
+// 			reports,
+// 			numOfContributors,
+// 		};
+// 	} catch (error) {
+// 		console.error("Failed to load reports or number of contributors:", error);
+// 		throw new Response("Failed to load data", { status: 500 });
+// 	}
+// };
+
+// let cacheData: IPageData;
+
+// export const clientLoader: ClientLoaderFunction = async ({ serverLoader }) => {
+// 	if (cacheData) {
+// 		return {
+// 			reports: cacheData.reports,
+// 			numOfContributors: cacheData.numOfContributors,
+// 		};
+// 	}
+
+// 	const serverLoaderData = await serverLoader<IPageData>();
+// 	const { reports, numOfContributors } = serverLoaderData;
+// 	cacheData = {
+// 		reports,
+// 		numOfContributors,
+// 	};
+// 	return cacheData;
+// };
+
+// clientLoader.hydrate = true;
+
+async function getData() {
+	const reports = await fetchReports();
+	const numOfContributors = await getNumberOfContributors();
+
+	return { reports, numOfContributors };
+}
+
+export default async function ReportsPage() {
+	const { reports, numOfContributors } = await getData();
+	const uniqueReports = new Set(reports);
+
+	// This throws an error because we are violating the rules of hooks.
```

Comment:
do you know what rule is being violated? i read [this](https://react.dev/warnings/invalid-hook-call-warning) and i don't see any of these problems. concerned that it may be a deeper issue with the React installation(?) bundling(?)

not sure where else to investigate this, and i also need access to `useMemo()` in this file for filtering.

when i remove `useMemo()` here and call `contributionAmounts()` in the components below that need the data, they are able to render. `ReportsHeader` subsequently uses several `useMemo`s wtihout issue.

** except ** 
- `reports-filters.tsx` needs 'use client' to render `Drawer` component (noted there)
- `MultipleSelector` component for states filter renders but mysteriously has no functionality (doesn't produce any errors)

Reply:
I didn't have time to debug it too much, still working on lots of other things. Can you check it out after the merge please?